### PR TITLE
Tests1 - category , book

### PR DESCRIPTION
**Pull request by dmytrocherepov**
https://ru.paste.pics/Q2M9V

https://ru.paste.pics/Q2MA3

### CODE REVIEW

**Review by OlehReex — APPROVED** [review]

File: src/main/java/com/example/bookstoreapp/model/Book.java [hunk 1]
```diff
@@ -40,7 +41,7 @@ public class Book {
     private String description;
     private String coverImage;
 
-    @ManyToMany
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/model/Book.java]: wow, don’t do it!!!

File: src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -57,14 +58,19 @@ public BookDto getById(Long id) {
 
     @Override
     public BookDto updateBookById(CreateBookRequestDto requestDto, Long id) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("No such book with id: " + id)
+        );
         Book book = bookMapper.toBook(requestDto);
+        setCategories(requestDto, book);
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java]: why do you duplicate it?

File: src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -57,14 +58,19 @@ public BookDto getById(Long id) {
 
     @Override
     public BookDto updateBookById(CreateBookRequestDto requestDto, Long id) {
+        bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("No such book with id: " + id)
+        );
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java]: you can use isBookExistById

File: src/test/java/com/example/bookstoreapp/service/BookServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceTest {
+
```
Comment (liliia-ponomarenko) [inline src/test/java/com/example/bookstoreapp/service/BookServiceTest.java]:
```suggestion
```

File: src/test/java/com/example/bookstoreapp/service/BookServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        book.setPrice(requestDto.price());
+        book.setCategories(Set.of(category));
+        book.setDescription(requestDto.description());
+        book.setCoverImage(requestDto.coverImage());
+        book.setTitle(requestDto.title());
+        book.setAuthor(requestDto.author());
+        book.setIsbn(requestDto.isbn());
+
+        when(bookRepository.findById(anyLong())).thenReturn(Optional.of(book));
+        when(bookMapper.toBook(requestDto)).thenReturn(book);
```
Comment (liliia-ponomarenko) [inline src/test/java/com/example/bookstoreapp/service/BookServiceTest.java]: do you really need to mock it?  you use spy

File: src/test/java/com/example/bookstoreapp/service/BookServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        Long id = -1L;
+        CreateBookRequestDto requestDto = new CreateBookRequestDto(
+                "checkTitle",
+                "checkAuthor",
+                "1234567890",
+                BigDecimal.valueOf(10),
+                null,
+                null,
+                List.of(1L)
+        );
```
Comment (liliia-ponomarenko) [inline src/test/java/com/example/bookstoreapp/service/BookServiceTest.java]: I see the same object in another method. You can move it to private method

File: src/test/java/com/example/bookstoreapp/service/CategoryServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    @Description("Saves a category")
+    void save_validCategoryRequestDto_returnsValidCategoryDto() {
+        CategoryRequestDto requestDto = new CategoryRequestDto(
+                "test",
+                "test"
+        );
+        Category category = new Category();
+        category.setName(requestDto.name());
+        category.setDescription(requestDto.description());
+        when(categoryMapper.toEntity(requestDto)).thenReturn(category);
```
Comment (liliia-ponomarenko) [inline src/test/java/com/example/bookstoreapp/service/CategoryServiceTest.java]: the same

File: src/test/java/com/example/bookstoreapp/service/CategoryServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        CategoryRequestDto requestDto = new CategoryRequestDto(
+                "test",
+                "test"
+        );
+        Category category = new Category();
+        category.setDescription(requestDto.description());
+        category.setName(requestDto.name());
+        category.setId(id);
+
+        when(categoryMapper.toEntity(requestDto)).thenReturn(category);
```
Comment (liliia-ponomarenko) [inline src/test/java/com/example/bookstoreapp/service/CategoryServiceTest.java]: the same

File: src/test/java/com/example/bookstoreapp/service/CategoryServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    @Description("Delete a category with existing id")
+    void deleteById_validId_ShouldDoNothing() {
+        Long id = 1L;
+
+        doNothing().when(categoryRepository).deleteById(id);
+        categoryService.deleteById(id);
+
+        verify(categoryRepository, times(1)).deleteById(id);
+    }
+
```
Comment (liliia-ponomarenko) [inline src/test/java/com/example/bookstoreapp/service/CategoryServiceTest.java]:
```suggestion
```

File: src/test/resources/databases/books/add-one-book.sql [hunk 10]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO books(id,title,author,isbn,price)
+values (1,'test','test','1234567890',1.25);
+
+INSERT INTO books_categories(book_id,category_id)
+values (1,1);
```
Comment (liliia-ponomarenko) [inline src/test/resources/databases/books/add-one-book.sql]: add one empty line (check other files)

File: src/test/resources/databases/books/add-one-book.sql [hunk 11]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO books(id,title,author,isbn,price)
+values (1,'test','test','1234567890',1.25);
```
Comment (liliia-ponomarenko) [inline src/test/resources/databases/books/add-one-book.sql]:
```suggestion
VALUES (1,'test','test','1234567890',1.25);
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! 👍 👏  Let’s  fix a few mistakes ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Great job!

**Pull request merged**